### DISCUSSION TITLE
How to change the default database port (5432)?

### DISCUSSION THREAD

**Discussion by Julien4135** [robske110/IDDataLogger#35] [Q&A] [answered]
Hello, is it possible please to change port 5432 that IDDataLogger uses easily ?
Because I have another application which uses this same port and which creates a conflict with me?

**Comment by robske110** [comment]
Port 5432 is used by postgresql. You should be able to change it in postgresql.conf and then in the config at [db.host](https://github.com/robske110/IDDataLogger/blob/master/config/config.example.json#L8) you'd need to add ;port=PORTNUMBER to localhost (the full string should be localhost;port=PORTNUM)

To give more specific help, I'd need to know what kind of setup you are running (which Operating System, what installation method did you use).

**Reply by Julien4135** [reply]
Good evening, I am using a Raspberry PI 3B + with this configuration:
 
By doing with the beginner version, and online with duckdns. on ios ;)

 pi@raspberrypi:~ $ cat /proc/version
Linux version 5.10.63-v7l+ (dom@buildbot) (arm-linux-gnueabihf-gcc-8 (Ubuntu/Linaro 8.4.0-3ubuntu1) 8.4.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #1488 SMP Thu Nov 18 16:15:28 GMT 2021

Thank you :)

**Reply by Julien4135** [reply]
By the way, I'm ID.3 on the SpeakEV forum ;)

**Reply by Julien4135** [reply]
Hello,

So there is an easier solution or I have to change the configuration in db.host?

**Reply by Julien4135** [reply]
{
    "username": "",
    "password": "",
    "vin": null,
    "base-updaterate": 600,
    "increased-updaterate": 60,
	"port=": 5422
    "db": {
        "host": "",
        "dbname": "",
        "user": "",
        "password": null,
        "driver": "pgsql"
    },
    "carpic": {
        "viewDirection": "front",
        "angle": "right",
        "flip": true
    },
    "timezone": null,
    "logging": {
        "debug-enable": false,
        "curl-verbose": false,
        "file-enable": false,
        "log-dir": null
    }
}

    © 2021 GitHub, Inc.

    Terms
    Privacy
    Security

**Reply by robske110** [reply]
What is the second application you are running on port 5432? As far as I know that port is only used by postgresql. There is no need to run a second postgresql server just for IDDataLogger, if you already have one running for another application.

If you still want to run the postgresql server on another port, you will need to configure it to run on that port (the config file might be at `/etc/postgresql/11/main/postgresql.conf` for you).
Then you can configure the port in the IDDataLogger config.json (`~/IDDataLogger/config/config.json`) file: (The only important part is the db.host line)
```
{
	"username": "",
	"password": "",
	"vin": null,
	"base-updaterate": 600,
	"increased-updaterate": 60,
	"db": {
		"host": "localhost;port=5422",
		"dbname": "",
		"user": "",
		"password": null,
		"driver": "pgsql"
	},
	"carpic": {
		"viewDirection": "front",
		"angle": "right",
		"flip": true
	},
	"timezone": null,
	"logging": {
		"debug-enable": false,
		"curl-verbose": false,
		"file-enable": false,
		"log-dir": null
	}
}
```

**Reply by Julien4135** [reply]
Hello, I'm actually using this project: https://github.com/tillsteinbach/VWsFriend
With your Widget it would complete the VWsFriend application;)

**Reply by tillsteinbach** [reply]
Actually you can change the port in VWsFriend quite easy by adding the DB_PORT variable to your env file.
The suggestion to use the same database is also good, but I don’t know if IDDataLogger can deal with Postgres 13?

**Reply by robske110** [reply]
What @tillsteinbach suggested is probably the easiest option. Alternatively, since I assume you are running VWsFriend using docker anyways, you could also just use the [docker](https://github.com/robske110/IDDataLogger/blob/master/docs/docker.md) version of IDDataLogger (although I don't recommend doing so on an older raspberry pi).
IDDataLogger can deal with PostgreSQL 13, it even works with 14 (there are usually no big breaking changes in PostgreSQL anyways).

Another thing: I'm not very well versed with docker, but I don't think the postgresql port should even be be bound on the host (where @Julien4135 was having problems with the port) with this configuration: https://github.com/tillsteinbach/VWsFriend/blob/main/docker-compose.yml#L27-L48
Is there something I am missing?

**Reply by Julien4135** [reply]
At the time I had to change the port on VWsFriend with the file that @tillsteinbach had to modify for me and it didn't want to work I don't know for what reason? I had doubted the port 5432 that you both use?
What I suggest if you want is that I install IDDataLogger tonight without changing the basic configuration and have then see if there is a conflict or not? Unless you're sure it doesn't work because of a possible port conflict?

**Reply by tillsteinbach** [reply]
Yes, in the "standard" docker-compose.yml the port is not exposed.
In the variant with HomeKit the port needs to be exposed. Unfortunately that is the only way to get multicast working...

**Reply by Julien4135** [reply]
Ok in addition it is precisely HomeKit that interests me

**Reply by Julien4135** [reply]
Otherwise it's really cool to see you both here trying to fix my problem! Really cool

**Reply by robske110** [reply]
I would recommend you to add the line `DB_PORT=5422` to the .env file (myconfig.env) of VWFriends, restart and then re-attempt installation of IDDataLogger. That should be the easiest option.

**Reply by tillsteinbach** [reply]
I hope it works I have to admit that I did not test it.

**Reply by Julien4135** [reply]
Ok I will keep you posted tomorrow morning following the installation that I am going to attempt this evening. I come home late from work around 10 p.m.
Again thanks to you !

**Reply by Julien4135** [reply]
Good evening so I put it like that in myconfig.env ?

VWSFRIEND_USERNAME=xxx
VWSFRIEND_PASSWORD=xxx
WECONNECT_USER=xxxx
WECONNECT_PASSWORD=xxxx
WECONNECT_PIN=xxxx # Pin is only necessary for some functions, if your car does not support these, the pin can be ignored
WECONNECT_INTERVAL=300
DB_USER=xxxx
DB_PASSWORD=xxxx
DB_PORT=5422
 If you don't want to use Apple Homekit you can ignore these settings
The following settings are only used with homekit in macvlan mode
HOMEKIT_INTERFACE=eth0
HOMEKIT_IP=192.168.0.234
HOMEKIT_MASK=192.168.0.0/24
HOMEKIT_GW=192.168.0.1

**Reply by Julien4135** [reply]
Apartment with this change everything works fine for the moment, after a reboot

**Comment by robske110** [comment, marked as answer]
This is how to change the port which is used for the postgresql database:

_Note1_: There is no need to run a second postgresql server just for IDDataLogger, if you already have one running for another application, you can just create a new empty database with a separate user and set that in IDDataLoggers config.

_Note2_: If you are using VWsFriend (this only applies for setups using the HomeKit docker file) it is easier to simply add `DB_PORT=5422` to the myconfig.env file.

If you still want to change the port IDDataLogger uses for its database, you will need to set db.host to `localhost;port=5422`. (You can replace localhost or the number 5422 if needed)
The relevant part of the config should look like this:
```
{
	...
	"db": {
		"host": "localhost;port=5422",
		...
	},
	...
}
```
(You'll also need to adjust the port setting in your relevant postgresql.conf, for debian/raspbian usually at `/etc/postgresql/13/main/postgresql.conf`)

**Comment by Julien4135** [comment]
Hello, I apologize for the time to answer but I finally had a lot of conflict (especially with duplicate postgresql servers) by doing my manipulation without really understanding. I came back to the same situation as last time and all of them are working fine on VWsFriend.

For _nota 1_ if I run the configuration for beginners with the command "" wget https://raw.githubusercontent.com/robske110/IDDataLogger/master/docs/install.sh; bash install.sh; rm install.sh "" I am going to create a second postgresql server, what not to do, if I understood correctly?
Do you have to do with the advanced, more complex solution?

_Nota 2_ I have modified the myconfig.env file from VWsFriend by modifying all the lines with 5432 and put 5422 in place, then reboot. For the moment no problem with the configuration with VWsFriend, it works as with the 5432 before.

So it remains to know how to create a new empty database with a separate user and define it in the IDDataLoggers configuration ;)

**Reply by robske110** [reply]
See my comment in https://github.com/robske110/IDDataLogger/discussions/39

**Reply by Julien4135** [reply]
![image](https://user-images.githubusercontent.com/93113346/147945945-b1a19469-6dfa-46ad-97f7-99910734c47a.png)

I still have the problem and I don't understand what to do with answer #39

**Reply by robske110** [reply]
Please post the output of the commands I posted in discussion #39.

**Reply by Julien4135** [reply]
![image](https://user-images.githubusercontent.com/93113346/147963615-7df9b575-ae5d-4008-bc11-1a80f3534c60.png)

**Reply by robske110** [reply]
Try to re-run the installation script.
`wget https://raw.githubusercontent.com/robske110/IDDataLogger/master/docs/install.sh; bash install.sh; rm install.sh`
You'll need to reenter username&password of the VW ID account and then answer N to overwriting defaults and Y to update existing configuration.
The script is designed to be run multiple times, it will not duplicate anything.

**Comment by Julien4135** [comment]
![image](https://user-images.githubusercontent.com/93113346/147965579-58913454-337a-49ae-b98b-0b2177e81200.png)

**Reply by robske110** [reply]
Does it start now?

**Reply by Julien4135** [reply]
pi@raspberrypi:~/IDDataLogger $ wget https://raw.githubusercontent.com/robske110/IDDataLogger/master/docs/install.sh; bash install.sh; rm install.sh
--2022-01-03 20:49:38--  https://raw.githubusercontent.com/robske110/IDDataLogger/master/docs/install.sh
Résolution de raw.githubusercontent.com (raw.githubusercontent.com)… 185.199.108.133, 185.199.111.133, 185.199.110.133, ...
Connexion à raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443… connecté.
requête HTTP transmise, en attente de la réponse… 200 OK
Taille : 1764 (1,7K) [text/plain]
Sauvegarde en : « install.sh »

install.sh                                                  100%[=========================================================================================================================================>]   1,72K  --.-KB/s    ds 0s      

2022-01-03 20:49:38 (4,98 MB/s) — « install.sh » sauvegardé [1764/1764]

Lecture des listes de paquets... Fait
Construction de l'arbre des dépendances... Fait
Lecture des informations d'état... Fait      
apt-transport-https est déjà la version la plus récente (2.2.4).
ca-certificates est déjà la version la plus récente (20210119).
curl est déjà la version la plus récente (7.74.0-1.3+deb11u1).
lsb-release est déjà la version la plus récente (11.1.0+rpi1).
0 mis à jour, 0 nouvellement installés, 0 à enlever et 4 non mis à jour.
--2022-01-03 20:49:40--  https://packages.sury.org/php/apt.gpg
Résolution de packages.sury.org (packages.sury.org)… 2606:4700:3030::ac43:b696, 2606:4700:3037::6815:1294, 172.67.182.150, ...
Connexion à packages.sury.org (packages.sury.org)|2606:4700:3030::ac43:b696|:443… connecté.
requête HTTP transmise, en attente de la réponse… 200 OK
Taille : 1769 (1,7K) [application/octet-stream]
Sauvegarde en : « /etc/apt/trusted.gpg.d/php.gpg »

/etc/apt/trusted.gpg.d/php.gpg                              100%[=========================================================================================================================================>]   1,73K  --.-KB/s    ds 0s      

2022-01-03 20:49:40 (5,56 MB/s) — « /etc/apt/trusted.gpg.d/php.gpg » sauvegardé [1769/1769]

Atteint :1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease
Atteint :2 http://archive.raspberrypi.org/debian bullseye InRelease                                                                                                        
Atteint :3 https://download.docker.com/linux/raspbian bullseye InRelease                                                                                                   
Atteint :4 https://packages.sury.org/php bullseye InRelease                                                                              
Lecture des listes de paquets... Fait
Lecture des listes de paquets... Fait
Construction de l'arbre des dépendances... Fait
Lecture des informations d'état... Fait      
Note : sélection de « php8.0-xml » au lieu de « php8.0-dom »
php8.0 est déjà la version la plus récente (8.0.14-1+0~20211220.28+debian11~1.gbpad68c7).
php8.0-curl est déjà la version la plus récente (8.0.14-1+0~20211220.28+debian11~1.gbpad68c7).
php8.0-gd est déjà la version la plus récente (8.0.14-1+0~20211220.28+debian11~1.gbpad68c7).
php8.0-pgsql est déjà la version la plus récente (8.0.14-1+0~20211220.28+debian11~1.gbpad68c7).
php8.0-xml est déjà la version la plus récente (8.0.14-1+0~20211220.28+debian11~1.gbpad68c7).
0 mis à jour, 0 nouvellement installés, 0 à enlever et 4 non mis à jour.
Lecture des listes de paquets... Fait
Construction de l'arbre des dépendances... Fait
Lecture des informations d'état... Fait      
postgresql est déjà la version la plus récente (13+225).
0 mis à jour, 0 nouvellement installés, 0 à enlever et 4 non mis à jour.
DO
createdb: erreur : la création de la base de données a échoué : ERREUR:  la base de données « vwid » existe déjà
Lecture des listes de paquets... Fait
Construction de l'arbre des dépendances... Fait
Lecture des informations d'état... Fait      
git est déjà la version la plus récente (1:2.30.2-1).
0 mis à jour, 0 nouvellement installés, 0 à enlever et 4 non mis à jour.
fatal: le chemin de destination 'IDDataLogger' existe déjà et n'est pas un répertoire vide.
Loading InteractiveWizard...
Loading ConfigWizard...
Welcome to the config wizard for the ID DataLogger!
This wizard makes it easy to enter the VW account login and database connection information into the respective config files.
You will now be asked for your VW account login information
> What is the username of your VW ID account?: xxxxxxx
> What is the password of your VW ID account?:xxxxxx
> A config.json already exists. Do you want to overwrite it with the defaults from config.example.json? (Y,N)
[N]: N
> Do you want to update your existing config with the parameters you entered in this wizard? (Answering N will exit this Wizard) (Y,N)
[Y]: Y
WARNING: FORCE_ALLOW_HTTP is enabled. This will prevent the frontend from blocking and redirecting http requests.
Make sure to disable this should you eventually enable https and expose this project to the internet!
To disable run ./config-wizard.sh --secure and make sure to replace the old .env file with the newly generated one!
rm: impossible de supprimer '/var/www/html/index.html': Aucun fichier ou dossier de ce type
mkdir: impossible de créer le répertoire « /var/www/html/vwid/ »: Le fichier existe
ln: impossible de créer le lien symbolique '/var/www/.env': Le fichier existe
Installation complete! You can now enter cd IDDataLogger && ./start.sh to finish setting up.
pi@raspberrypi:~/IDDataLogger $

**Reply by robske110** [reply]
Sorry if I wasn't clear, I wanted to know if IDDataLogger is now able to start without the error. (Run `./start.sh`)

**Reply by Julien4135** [reply]
Ok,

![image](https://user-images.githubusercontent.com/93113346/148007513-c7aeaf37-d365-42b3-b9ea-20a3f6a74d7d.png)

**Reply by Julien4135** [reply]
For information, I have local access to the login page.

![image](https://user-images.githubusercontent.com/93113346/148013369-016dc117-2e27-4acc-86a4-d5ac606bbb87.png)

**Reply by robske110** [reply]
Try to manually set the db password:
```
cd ~/IDDataLogger
sudo su postgres
SELECT rolname FROM pg_roles;
ALTER ROLE iddatalogger WITH PASSWORD '1234';
exit
exit
nano config/config.json
```
The double exit is intentional!
Once you're in nano (last command), navigate to the db.password line and set it to 1234. `"password": "1234",`.
Then press CTRL+O (^O) and CTRL+X (^X).
Try starting now. If it still failed, it is likely that IDDataLogger is somehow still connecting to the wrong database server.

**Reply by Julien4135** [reply]
ok I don't know if it's like that, but it doesn't work yet :( 

![image](https://user-images.githubusercontent.com/93113346/148042343-6865c5f1-9be6-4f83-9dbf-c61450826457.png)

![image](https://user-images.githubusercontent.com/93113346/148042854-ae72d80a-f47c-4703-9c65-08c2eea60541.png)

**Reply by robske110** [reply]
You need to enter the commands I sent one-by-one. Also, the db.user field should still be iddatalogger.

**Reply by Julien4135** [reply]
Unable to connect 

![image](https://user-images.githubusercontent.com/93113346/148043304-28290292-3250-4f9c-a1ae-5d7bf06cf998.png)

**Reply by robske110** [reply]
You need to enter these commands one-by-one (one line at a time, followed by enter):
```
cd ~/IDDataLogger
sudo su postgres
SELECT rolname FROM pg_roles;
ALTER ROLE iddatalogger WITH PASSWORD '1234';
exit
exit
nano config/config.json
```
And in the config db.user needs to be "iddatalogger".

**Reply by Julien4135** [reply]
![image](https://user-images.githubusercontent.com/93113346/148044719-e8ee7289-7259-4680-a994-2744a3f0713a.png)

**Reply by robske110** [reply]
Oh, sorry the commands should've been
```
cd ~/IDDataLogger
sudo su postgres
psql
ALTER ROLE iddatalogger WITH PASSWORD '1234';
exit
exit
nano config/config.json
```

**Reply by Julien4135** [reply]
Here I have to do what now ./start.sh  ? 

![image](https://user-images.githubusercontent.com/93113346/148047473-f4132202-0201-4cd2-9436-a1526f4d8b1b.png)

![image](https://user-images.githubusercontent.com/93113346/148047625-c3c28c80-78da-4874-9620-05d200b57a2b.png)

**Reply by robske110** [reply]
Yes, try to run ./start.sh

**Reply by Julien4135** [reply]
It does not work 

![image](https://user-images.githubusercontent.com/93113346/148048466-0538ded2-3861-4652-894c-ce4623c3cb08.png)

**Reply by robske110** [reply]
Then the database of VW's friend still appears to be running on port 5432.

Run `pg_isready` and post the output here.

Try to change the port of the postgresql database now:
```
sudo nano /etc/postgresql/13/main/postgresql.conf
```
Go to line 64 and change it to `PORT = 5464`.

Then you need to set db.host to localhost;port=5464 in the config.json of IDDataLogger
The relevant part of the config should look like this:
```
{
	...
	"db": {
		"host": "localhost;port=5464",
		...
	},
	...
}
```

Then you'll need to restart your raspberrypi: `sudo reboot`
Run `pg_isready` again and post the output here.
Then run `./start.sh` and see if it works now.

**Reply by Julien4135** [reply]
![image](https://user-images.githubusercontent.com/93113346/148055546-d62b1edd-5b17-4c53-81df-413f53c30428.png)

**Reply by Julien4135** [reply]
Sorry I do not have to understand everything, I have the impression that it will never work :(

**Reply by robske110** [reply]
You just need to execute the pg_isready command in a normal terminal. (without executing sudo su postgres and psql).
Then just follow the steps in my last post one-by-one.

**Reply by Julien4135** [reply]
Ok 

![image](https://user-images.githubusercontent.com/93113346/148093964-b0675ea4-a69c-4f70-bfb2-158b66de3f25.png)

I continue

**Reply by Julien4135** [reply]
![image](https://user-images.githubusercontent.com/93113346/148095426-fc4f688e-cb02-4ce9-978c-3c404ba62cf4.png)

**Reply by Julien4135** [reply]
By restarting now on VW Friend it does not work anymore, until now with all the modifications without reboot everything worked

In fact the raspberry is completely blocked I can not even open a terminal window

**Reply by Julien4135** [reply]
I will have to start from scratch

**Reply by robske110** [reply]
None of the actions should've done anything meaningful to the operating system. What exactly does not work now?

**Reply by Julien4135** [reply]
when I click on the terminal to open a window the navigation bar disappears and comes back a few seconds later and also impossible to access the raspberry menu by clicking nothing happens

**Reply by Julien4135** [reply]
I think my equipment is getting old yet it is a PI3 b + of 2 or 3 years max

**Reply by Julien4135** [reply]
Or rather the micro SD card which starts to give back the soul

**Reply by robske110** [reply]
No, you just have a weird raspbian UI bug, it should not have anything to do with hardware (only maybe SD card data corruption).
If a reboot doesn't help, you can try to boot into single user mode. (There are plenty of guides online on how to do this)

**Reply by Julien4135** [reply]
no I will reinstall, I will start with IDDataloger with a blank config without VWsFriend for the moment.

**Reply by Julien4135** [reply]
I will be back in about an hour, thank you for your help

**Reply by Julien4135** [reply]
As usual, I am able to flash the micro sd card (a new one) when I run the first configuration, it is impossible to write on drive E: or another letter I don't know anymore, from the terminal impossible to do the updated by usual orders.

I think my raspberry is fucked up

**Reply by robske110** [reply]
what first configuration are you referring to? Just use the raspberry pi imager from here https://www.raspberrypi.com/software/

**Reply by Julien4135** [reply]
this is exactly what I use

**Reply by Julien4135** [reply]
If you are still available here for an hour or so, can I show you a photo of the errors during the update?

**Reply by robske110** [reply]
yes

**Reply by Julien4135** [reply]
ok

**Reply by Julien4135** [reply]
![11](https://user-images.githubusercontent.com/93113346/148121147-f02e2d0f-21b7-48e7-a714-fd58440fd4dd.jpg)

**Reply by Julien4135** [reply]
It's serious ?

**Reply by Julien4135** [reply]
![image](https://user-images.githubusercontent.com/93113346/148121962-20dcc814-9ecb-4819-b0e8-4dae6702a96e.png)

**Reply by robske110** [reply]
That's very weird. (But it certainly is not any hardware issue)
Maybe the mirror (igh.cnrs.fr) is somehow messed up.
Did you download a fresh image?
I would try to run
```
sudo apt update
sudo apt full-upgrade --fix-missing
```
if that still doesn't work try manually selecting a raspbian mirror.
See the list here:
https://www.raspbian.org/RaspbianMirrors
You can google how to change the mirror (you'll need to edit `sudo nano /etc/apt/sources.list`)

**Reply by Julien4135** [reply]
ok i try

**Reply by Julien4135** [reply]
for the moment...

![image](https://user-images.githubusercontent.com/93113346/148123443-e63b0f88-081a-43a0-be7a-18e627529380.png)

**Reply by Julien4135** [reply]
It's progressing...

![image](https://user-images.githubusercontent.com/93113346/148123788-55810da9-50cc-41d2-b058-871bb7caf8f2.png)

**Reply by Julien4135** [reply]
I think it's not good? always this error with drive E: 

![image](https://user-images.githubusercontent.com/93113346/148124613-df2f8d9a-ecec-4c67-a046-f6a2726f9d0e.png)

**Reply by robske110** [reply]
That's not an error with a drive (this is not windows!). E: just literally means Error:
I think you can ignore it for now. It is either a repository issue or mirror issue and doesn't really matter. Try to run the commands again in a day or so and see if it still happens.

You can now continue to install the applications you need.

**Reply by Julien4135** [reply]
ok i try to get IDDataLogger working first

**Reply by Julien4135** [reply]
There really is a problem with E: I think?

![image](https://user-images.githubusercontent.com/93113346/148125865-c55dc982-cb69-4220-a591-5c5b9f2dfd4e.png)

**Reply by robske110** [reply]
Then you should really look into changing the repository mirror.
See the list here:
https://www.raspbian.org/RaspbianMirrors
You can google how to change the mirror (you'll need to edit sudo nano /etc/apt/sources.list)

Was this a freshly downloaded image?

**Reply by Julien4135** [reply]
I used the official raspberry software for the image of Os, so yes it is the current image that is offered first on their software.
I will try another installation tomorrow.
Thanks again and see you tomorrow

**Reply by Julien4135** [reply]
It's good the installation I made of the raspberry bone is working and I'm up to date.
So I start by installing IDDataLogger first

**Reply by Julien4135** [reply]
So all seems well installed but after having launched ./start.sh I had the API key I created a user but I do not have the proposal for the password.

Here is what I got from user input:

Do you want to create an user? (Y,N)
[Y]: XXXXXX
Please answer with one of the following options (case-sensitive!): Y,N
> Do you want to create an user? (Y,N)

I answered YES

**Reply by robske110** [reply]
Just re-run the wizard with `./start.sh --wizard`. You need to answer with an uppercase Y to the `Do you want to create a user?` question.

**Reply by Julien4135** [reply]
Great ! it works locally for the moment

I continue to upload

**Reply by robske110** [reply]
Great! I also fixed the small bug you encountered with the wizard here: https://github.com/robske110/IDDataLogger/commit/705d3b2cc97a7d2a941d98bcaacf05c8ad6614f4

**Reply by Julien4135** [reply]
Nickel!

**Reply by Julien4135** [reply]
It's won I managed to meter online as before and everything works as before! Really cool !!!
Is it now complicated to just change the port or do things have to be done at the risk of it not working anymore?

**Reply by Julien4135** [reply]
Or do I go directly to the installation of VWsFriend?

**Reply by robske110** [reply]
You can try to change the port of the IDDataLogger database now, if it doesn't work you can always undo your changes.

sudo nano /etc/postgresql/13/main/postgresql.conf
Go to line 64 and change it to PORT = 5464.

Then you need to set db.host to localhost;port=5464 in the config.json of IDDataLogger
The relevant part of the config should look like this:

{
	...
	"db": {
		"host": "localhost;port=5464",
		...
	},
	...
}
Then you'll need to restart your raspberrypi: sudo reboot
Run `pg_isready` and post the output here.
Then run ./start.sh and see if it still works.
